datalad_dataverse/tests/test_dataset.py

```diff
@@ -19,30 +19,42 @@ def test_file_handling(
         dataverse_dataaccess_api,
         dataverse_dataset,
 ):
-    # the starting point of `dataverse_dataset` is a freshly
-    # created, non-published dataset in draft mode, with no prior
-    # version
     odd = ODD(dataverse_admin_api, dataverse_dataset)
 
-    fcontent = 'some_content'
-    fpath = tmp_path / 'dummy.txt'
-    fpath.write_text(fcontent)
-    src_md5 = md5sum(fpath)
-
-    fileid = check_upload(odd, fcontent, fpath, src_md5)
-
-    check_download(odd, fileid, tmp_path / 'downloaded.txt', src_md5)
-
-    check_file_metadata_update(dataverse_admin_api, dataverse_dataset, odd,
-                               fileid, fpath)
-
-    fileid = check_replace_file(odd, fileid, tmp_path)
-
-    check_rename_file(odd, fileid)
+    paths = (
+        tmp_path / '.Ö-dot-Ö-in-front' / '!-üö-.txt',
+        tmp_path / '.dot-in-front' / 'c1.txt',
+        tmp_path / ' space-in-front' / 'c2.txt',
+        tmp_path / '-minus-in-front' / 'c3.txt',
+        tmp_path / 'Ö-in-front' / 'überflüssiger_fuß.txt',
+        tmp_path / ' Ö-space-Ö-in-front' / 'a.txt',
+        tmp_path / 'dummy.txt',
+    )
 
-    check_remove(odd, fileid, PurePosixPath(fpath.name))
+    path_info = dict()
+    for path in paths:
+        if path.parent != tmp_path:
+            path.parent.mkdir()
+        relative_path = path.relative_to(tmp_path)
+        fcontent = 'content of: ' + str(relative_path)
+        path.write_text(fcontent)
+        src_md5 = md5sum(path)
+        fileid = check_upload(odd, fcontent, relative_path, tmp_path, src_md5)
+        path_info[path] = (src_md5, fileid)
 
-    check_duplicate_file_deposition(odd, tmp_path)
+    for path, (src_md5, fileid) in path_info.items():
+        relative_path = path.relative_to(tmp_path)
+        check_download(odd, fileid, tmp_path / 'downloaded.txt', src_md5)
+        check_file_metadata_update(
+            dataverse_admin_api, dataverse_dataset, odd, fileid, path
+        )
+        path_to_check = relative_path.parent / 'mykey'
+        fileid = check_replace_file(odd, fileid, path_to_check, tmp_path)
+        check_rename_file(
+            odd, fileid, name=str(relative_path.parent / ('ren' + path.name))
```

Comment:
```suggestion
            odd, fileid, name=PurePosixPath(relative_path.parent / ('ren' + path.name))
```

Reply:
otherwise, this creates a path mess on Windows (ref Michaels comment in the chat) that ultimately turns into an HTTP 500 error

Reply:
Good suggestion, thx